examples/lzma2_reader.rs

```diff
@@ -0,0 +1,18 @@
+use std::io::{self, Read, Write};
+
+use lzma_rust2::{Lzma2Options, Lzma2Reader, Lzma2Writer, LzmaOptions};
+
+fn main() -> io::Result<()> {
+    let input = b"Hello, world!";
+
+    let mut writer = Lzma2Writer::new(Vec::new(), Lzma2Options::default());
```

Comment:
Please adjust the LZMA2 writers more to the other examples. Let the LZMA2 writer output a file, that that the LZMA2 reader can read. You only need to preprend the dictionary size to the LZMA2 data, which later then LZMA2 reader will read before creating the reader.